jenkins/scripts/run_tests.py

```
@@ -0,0 +1,705 @@
+        pytest_cmd: The base pytest command string.
+        isolate_list: Path to the isolate test list file.
+        working_dir: Working directory for pytest execution.
+        output_dir: Output directory for the stage.
+        stage_name: Name of the test stage.
+        fail_signatures: List of failure signature strings.
+        max_rerun_tests: Max number of failed tests that can trigger rerun.
+
+    Returns:
+        Tuple of (rerun_failed: bool, all_xml_files: list of result XML paths).
+    """
+    isolate_tests = [l.strip() for l in Path(isolate_list).read_text().splitlines() if l.strip()]
```

Ruff (0.15.6)

[error] Ambiguous variable name: `l`

(E741)